### DISCUSSION TITLE
QST:how to use within func with set tolerance = 1e-8

### DISCUSSION THREAD

**Discussion by yyandgg** [geopandas/geopandas#2729] [Q&A]
hello, i have a question, can i set tolerance  when use within func or contains func

**Comment by ccalvocm** [comment]
Greetings @yyandgg 

You can set a tolerance when using the within or contains functions in GeoPandas. Both of these functions take a buffer argument that allows you to specify a tolerance distance.

Here's an example of how to use the within function with a tolerance:

```
import geopandas as gpd
from shapely.geometry import Point

# Create a GeoDataFrame with a polygon
polygons = gpd.GeoDataFrame({'geometry': [Point(0, 0).buffer(1)]})

# Create a point that is within the polygon, but outside the tolerance distance
point = Point(0.5, 0.5)

# Check if the point is within the polygon with a tolerance of 0.1
is_within = polygons.geometry.within(point.buffer(0.1))

print(is_within)
```

In this example, we create a GeoDataFrame with a single polygon that has a radius of 1 unit. We then create a point that is within the polygon, but outside the tolerance distance of 0.1 units. We use the within function with a buffer of 0.1 units to check if the point is within the polygon with a tolerance. The result is True, because the point is within the polygon with a tolerance of 0.1 units.

You can use a similar approach with the contains function to set a tolerance distance.
